src/persist-client/src/read.rs

```diff
@@ -689,17 +690,30 @@ where
     /// The `Since` error indicates that the requested `as_of` cannot be served
     /// (the caller has out of date information) and includes the smallest
     /// `as_of` that would have been accepted.
-    #[instrument(level = "debug", skip_all, fields(shard = %self.machine.shard_id()))]
-    pub async fn listen(self, as_of: Antichain<T>) -> Result<Listen<K, V, T, D>, Since<T>> {
```

Comment:
I'm a bit sad that we're changing signatures here... in particular it seems brittle if this method is refactored.

IIUC it would be ~nearly as good to insert a `.boxed()` before the `.await` and leave the signature alone... is that right, and if so how would you feel about it?

Reply:
We definitely can revert the change to the signature!

The only reason I originally made the changes instead of using `.boxed()` was because certain functions were called numerous times, and it seemed easier to change the signature rather than all of the callsites.

Reply:
Yeah +1 on this too, I dislike baking this into signatures. As long as the lint catches it if the boxed call is missed at a callsite, then I have a decently strong preference to do that everywhere in the persist bits of this change.